crates/bevy_reflect/src/serde/de.rs

```diff
@@ -538,6 +560,60 @@ impl<'a, 'de> Visitor<'de> for StructVisitor<'a> {
     }
 }
 
+struct NewtypeStructVisitor<'a> {
+    tuple_struct_info: &'static TupleStructInfo,
+    registry: &'a TypeRegistry,
+    registration: &'a TypeRegistration,
+}
+
+impl<'a, 'de> Visitor<'de> for NewtypeStructVisitor<'a> {
+    type Value = DynamicTupleStruct;
+
+    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+        formatter.write_str("reflected newtype struct value")
+    }
+
+    fn visit_newtype_struct<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let mut tuple_struct = DynamicTupleStruct::default();
+
+        let ignored_len = self
+            .registration
+            .data::<SerializationData>()
+            .map(|data| data.len())
+            .unwrap_or(0);
+        let field_len = self
+            .tuple_struct_info
+            .field_len()
+            .saturating_sub(ignored_len);
+
+        if field_len > 1 {
+            return Err(Error::custom(format_args!(
+                "Tried to deserialize a struct {} with more than 1 field as a newtype.",
+                self.tuple_struct_info.type_name(),
+            )));
+        }
+
+        let field = self.tuple_struct_info.field_at(0).ok_or_else(|| {
+            de::Error::custom(format_args!(
+                "no fields on supposedly newtype struct {}",
```

Comment:
Nit: I don't think `supposedly` is needed
```suggestion
                "no fields on newtype struct {}",
```